### ISSUE TITLE
Filtered format: Recognize a table view class=datatable

### ISSUE THREAD

**Issue by krabina** [SemanticMediaWiki/SemanticResultFormats#246]
the current version of SRF 2.5.0 can have multiple views, e. g. swichting from calendar to list of from map to table. 

Another great feature might be to introduce an "export" view that enables a third tab with export formats to download the displayed data based on the selected filters.

|export format=csv,json
could then result in a CSV and JSON link for data download.

**Comment by krabina** [comment]
I suggest to change this to bug, and reframe it:
"Filtered format  does not recognize table view class=datatable"

In  SMW 3.0 there is the  class=datatable feature, seehttps://www.semantic-mediawiki.org/wiki/Help:Table_format

however, `|table view class=datatable compact`
is not recognized, you can see it here: https://www.verwaltungspreis.gv.at/TestTable

If filtered format supported the  datatables class it would remove the need for an export view.

**Comment by krabina** [comment]
https://sandbox.semantic-mediawiki.org/wiki/Issue/SRF/246

MediaWiki | 1.32.0-rc.1 (9633c6d)11 décembre 2018 à 17:54
PHP | 7.0.30-0+deb9u1 (apache2handler)
MariaDB | 10.1.37-MariaDB-0+deb9u1
ICU | 57.1
LuaSandbox | 3.0.3
Lua | 5.1.5
Elasticsearch | 6.5.0

Semantic Result FOrmats | 3.1.0-alpha (0012812) 29 novembre 2018 à 17:59

**Comment by kghbln** [comment]
>  "Filtered format does not recognize table view class=datatable"

Makes sense since this class comes with an export feature.

> I suggest to change this to bug, and reframe it:

I do not think this is a bug since this is about honoring a new feature, i.e. there is no regression to previous behaviour.

@s7eph4n FYI and comment

**Comment by krabina** [comment]
FYI: class=datatable is specified in 
https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/2420#issuecomment-294453317

**Comment by s7eph4n** [comment]
To be noted that Filtered and Tables are completely different formats. A Filtered view is not just some other format with a bit of decoration. Any feature that another format might have would have to be recreated from scratch to be available for Filtered.

**Comment by krabina** [comment]
yes, I understand. So basically we have to discuss the relation of filtered with datatables. My problem is that I never could use the datatables format much, because of bugs that have existed ever since the format was implemented. 
Datatables is not only comparable to the tables view in filtered, (datatables also allows filtereing), but there are two distinct features of datatables that are lacking in filtered format:

1. Export button. This could be implemented with an "export" view in filtered
2. Search in results. This could also be implemented with a search view in filtered.

I just thought that it could be the easier way to use the datatables framework for the tables view and therefore it might be easier to achieve these two features beause they are already there.

But I can see the point that it is the better option to implement whatever is missing in filtered format there not to be dependent on any other result format.
